mesonbuild/interpreter/compiler.py

```diff
@@ -591,6 +591,14 @@ def find_library_method(self, args: T.Tuple[str], kwargs: 'FindLibraryKW') -> 'd
             if not self._has_header_impl(h, has_header_kwargs):
                 return self.notfound_library(libname)
 
+        # If has_headers is enabled and header_include_directories were passed,
+        # then construct compile_args from those include dirs.
+        compile_args = []
+        if kwargs['has_headers']:
+            for incdirs in kwargs['header_include_directories']:
+                for arg in incdirs.to_string_list(self.environment.get_source_dir()):
+                    compile_args.append(f'-I{arg}')
```

Comment:
I think we should store them as `IncludeDirs` objects on `ExternalProgram` object. Its base class already has `get_include_dirs()` that we can implement to return that list.